src/liballoc/boxed.rs

```diff
@@ -101,6 +101,45 @@ impl<T> Box<T> {
     }
 }
 
+impl<T : ?Sized> Box<T> {
+    /// Constructs a box from the raw pointer.
+    ///
+    /// After this function call, pointer is owned by resulting box. In
+    /// particular, it means that `Box` destructor calls destructor of
+    /// `T` and releases memory.
+    ///
+    /// Function is unsafe, because improper use of this function may
+    /// lead to memory problems like double-free, for example if the
+    /// function is called twice on the same raw pointer.
+    #[unstable]
+    pub unsafe fn from_raw(raw: *mut T) -> Self {
+        mem::transmute(raw)
+    }
+}
+
+/// Consumes the `Box`, returning the wrapped raw pointer.
+///
+/// After call to this function, caller is responsible for the memory
+/// previously managed by `Box`, in particular it is caller should
+/// properly destroy `T` and release memory.
```

Comment:
You may want to mention here that the "proper method" of releasing the memory is calling `Box::from_raw` at some point.